### PR TITLE
 [opencv.js benchmark] Adapt the benchmark to web browser

### LINKED ISSUE #263: Adapt the benchmark to web browser
> Found now nodejs don't support SharedArrayBuffer which is required for WASM threads. We may need another version of benchmark for html page(run in the brower like Chrome)

https://github.com/huningxin/opencv/pull/262#issuecomment-505286862

### PR DESCRIPTION
**Pull request by Wenzhao-Xiang**
Fix #263 

### This pullrequest changes
Made a simple adaptation of  node.js and web browser  for opencv.js benchmark.

### CODE REVIEW

**Comment by huningxin** [comment]
@Wenzhao-Xiang , there are commits in your previous PR. Could you please rebase your branch to `gsoc_2019` of this repo and upload again.

File: modules/js/perf/perf_imgproc/perf_cvtcolor.js [hunk 1]
```diff
@@ -461,60 +475,105 @@ cv.onRuntimeInitialized = () => {
...
+      } else {
+        log("no filter or getting invalid params, run all the cases");
+        addCvtModeCase(suite, combiCvtMode);
+        addCvtModeBayerCase(suite, combiCvtModeBayer);
+        addCvtMode2Case(suite, combiCvtMode2);
+        addCvtMode3Case(suite, combiCvtMode3);
+      }
+      setBenchmarkSuite(suite);
+      log(`Running ${totalCaseNum} tests from CvtColor`);
+      suite.run({ 'async': true }); // run the benchmark
```
Comment (huningxin) [inline modules/js/perf/perf_imgproc/perf_cvtcolor.js]: I suggest to combine the common code for Node.js and Browser from `suite = new Benchmark.Suite;` to `suite.run({ 'async': true });`. Please avoid the code duplication as much as possible.

Comment (Wenzhao-Xiang) [reply modules/js/perf/perf_imgproc/perf_cvtcolor.js]: Yeah. I abstracted the same code into a function named `genBenchmarkCase`. There are some difference between two version so it seems that these two part can't be combined totally.

**Review by huningxin — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**